flycheck.el

```diff
@@ -3685,7 +3690,16 @@ Intended for use with `next-error-function'."
   (let ((pos (flycheck-next-error-pos n reset)))
     (if pos
         (goto-char pos)
-      (user-error "No more Flycheck errors"))))
+      (let ((result (cl-remove-if
+                     (lambda (err)
+                       (eq (current-buffer)
+                           (flycheck-error-buffer err)))
+                     flycheck-current-errors)))
+        (if result
+            (let ((error (car result)))
+              (flycheck-jump-to-error error)
+              (flycheck-buffer))
+          (user-error "No more Flycheck errors"))))))
```

Comment:
Well, flycheck tries to jump to the next error. If there are none left in this buffer, then it puts in `result` only errors from other buffers instead. If there is one to jump to, then it jumps to it and re-checks (because each buffer has its own error list, we should re-check to ensure consistency with the current set of errors for a smoother UX).